pages/board/index.tsx

```diff
@@ -58,16 +48,18 @@ export default function Boards() {
         <section className={styles["boards-list"]}>
           <div className={styles["list-wrapper-title"]}>
             <h2 className={styles["boards-list-h2"]}>게시글</h2>
+            <Link href="/addboard">
             <button type="button" className={styles.boardsBtn}>
               글쓰기
             </button>
+            </Link>
           </div>
 
           <div className={styles["list-wrapper-input"]}>
-            <SearchForm initialValue={(q as string) || ""} />
+            <SearchForm initialValue={(keyword as string) || ""} />
```

Comment:
위에서 keyword에 대한 여러가지 경우의 수를 고려하여 이미 처리를 해주셨으므로
keyword 대신 searchKeyword를 사용해 주시면 별도의 타입 단언이나 `||`를 사용할 필요 없이 간단히 작성할 수 있을 것 같습니다.
```suggestion
            <SearchForm initialValue={searchKeyword} />
```